From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Erik Corry <erikcorry@chromium.org>
Date: Mon, 3 Mar 2025 22:30:37 +0100
Subject: Modify where to look for fast_float and simdutf.

Similar to fp16, these dependencies now needs to be downloaded by bazel.
The external @simdutf and @fast_float packages use include_prefix to
provide headers at the upstream paths (third_party/simdutf/simdutf.h and
third_party/fast_float/src/include/fast_float/fast_float.h), so no C++
include changes are needed.

Signed-off-by: James M Snell <jsnell@cloudflare.com>

diff --git a/BUILD.bazel b/BUILD.bazel
index 2590737339fa2cd939bcecd029f44ef6e95a305d..f269b7359344a47cfa48b665ef10b6c3da657907 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -4566,7 +4566,6 @@ filegroup(
         # than a blanket third_party/**, which would also stage the
         # large gclient trees.
         "third_party/utf8-decoder/**/*.h",
-        "third_party/simdutf/**/*.h",
         "third_party/rapidhash-v8/**/*.h",
         "third_party/wasm-api/**/*.h",
         "third_party/siphash/**/*.h",
@@ -4677,7 +4676,7 @@ run_metagen(
         ":define_flags",
         ":v8_libbase",
         ":lib_fp16",
-        ":simdutf",
+        "@simdutf",
         "@abseil-cpp//absl/container:btree",
         "@abseil-cpp//absl/container:flat_hash_map",
         "@abseil-cpp//absl/container:flat_hash_set",
@@ -4925,18 +4924,19 @@ cc_library(
     ],
 )
 
-cc_library(
-    name = "simdutf",
-    srcs = ["third_party/simdutf/simdutf.cpp"],
-    hdrs = ["third_party/simdutf/simdutf.h"],
-    strip_include_prefix = "third_party/simdutf",
-    copts = select({
-        "@v8//bazel/config:is_clang": ["-std=c++20"],
-        "@v8//bazel/config:is_gcc": ["-std=gnu++2a"],
-        "@v8//bazel/config:is_windows": ["/std:c++20"],
-        "//conditions:default": [],
-    }),
-)
+# The simdutf library is commented out to avoid conflicts with
+# the version we use in workers
+# cc_library(
+#     name = "simdutf",
+#     srcs = ["third_party/simdutf/simdutf.cpp"],
+#     hdrs = ["third_party/simdutf/simdutf.h"],
+#     copts = select({
+#         "@v8//bazel/config:is_clang": ["-std=c++20"],
+#         "@v8//bazel/config:is_gcc": ["-std=gnu++2a"],
+#         "@v8//bazel/config:is_windows": ["/std:c++20"],
+#         "//conditions:default": [],
+#     }),
+# )
 
 v8_library(
     name = "v8_libshared",
@@ -4983,15 +4983,15 @@ v8_library(
     }),
     deps = [
         ":lib_dragonbox",
-        "//third_party/fast_float/src:fast_float",
         ":lib_fp16",
-        ":simdutf",
         ":v8_libbase",
         "@abseil-cpp//absl/container:btree",
         "@abseil-cpp//absl/container:flat_hash_map",
         "@abseil-cpp//absl/container:flat_hash_set",
         "@abseil-cpp//absl/functional:overload",
         "@highway//:hwy",
+        "@fast_float",
+        "@simdutf",
     ],
 )
 
